Star patterns in `match` statement aren't captured

The following program:

```python
match [123]:
    case [*x]:
        print(x)

match {"abc": 123}:
    case {**x}:
        print(x)
```

Emits the following errors:

```
$ ruff file.py
file.py:3:15: F821 Undefined name `x`
file.py:7:15: F821 Undefined name `x`
```

Currently using the latest version:

```
$ ruff --version
ruff 0.0.252
```

--- a/crates/ruff/resources/test/fixtures/pyflakes/F821_9.py
+++ b/crates/ruff/resources/test/fixtures/pyflakes/F821_9.py
@@ -20,7 +20,27 @@ def f(provided: int) -> int:
     match provided:
         case True:
             return captured  # F821
+
+
+def f(provided: int) -> int:
+    match provided:
+        case captured:
+            return captured
+
+
+def f(provided: int) -> int:
+    match provided:
         case [captured, *_]:
             return captured
-        case captured:
+
+
+def f(provided: int) -> int:
+    match provided:
+        case [*captured]:
+            return captured
+
+
+def f(provided: int) -> int:
+    match provided:
+        case {**captured}:
             return captured

--- a/crates/ruff/src/checkers/ast.rs
+++ b/crates/ruff/src/checkers/ast.rs
@@ -3863,6 +3863,10 @@ where
     fn visit_pattern(&mut self, pattern: &'b Pattern) {
         if let PatternKind::MatchAs {
             name: Some(name), ..
+        }
+        | PatternKind::MatchStar { name: Some(name) }
+        | PatternKind::MatchMapping {
+            rest: Some(name), ..
         } = &pattern.node
         {
             self.add_binding(

